File: BackEnd/Carpooling.Net/CarpoolingNET/Models/Triphistory.cs
using System;
using System.Collections.Generic;

namespace CarpoolingNET.Models
{
    public partial class Triphistory
    {
        public int TripId { get; set; }
        public int RideId { get; set; }
        public int Uid { get; set; }
        public int? Rating { get; set; }
        public string? Feedback { get; set; }

        public virtual Ride Ride { get; set; } = null!;
        public virtual User UidNavigation { get; set; } = null!;
    }
}

File: BackEnd/CarPooling.Java/src/main/java/com/example/demo/entities/Triphistory.java
package com.example.demo.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table(name="triphistory")
public class Triphistory {
@Id
@GeneratedValue(strategy=GenerationType.IDENTITY)
int tripid;

@JsonIgnoreProperties("trips")
@ManyToOne(cascade = CascadeType.ALL)
@JoinColumn(name="rideid")
Ride rideid;

int rating;
String feedback;
}
